test/annotation/utils_tests.cxx

```diff
@@ -0,0 +1,93 @@
+#define BOOST_TEST_MODULE annotation_utility_tests
+#include <unity/lib/annotation/image_classification.hpp>
+
+#include <unity/lib/gl_sarray.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include <util/test_macros.hpp>
+
+#include <sframe/testing_utils.hpp>
+#include <util/testing_utils.hpp>
+
+#include <image/image_type.hpp>
+
+#include "utils.cpp"
+
+struct utils_test {
+public:
+  /*
+   * Test Pass Through
+   *
+   * This test is supposed to check that the data that gets passed into the
+   * annotation utility gets passed out in the same manner with the same data
+   * format.
+   *
+   */
+
+  void test_featurize_images() {
+    std::string image_column_name = "image";
+    std::string annotation_column_name = "annotate";
+    std::shared_ptr<turi::unity_sframe> annotation_sf =
+        annotation_testing::random_sframe(50, image_column_name,
+                                          annotation_column_name);
+
+    std::shared_ptr<turi::unity_sarray> image_sarray =
+        std::static_pointer_cast<turi::unity_sarray>(
+            annotation_sf->select_column(image_column_name));
+
+    std::shared_ptr<turi::gl_sarray> image_gl_sarray =
+        std::make_shared<turi::gl_sarray>(image_sarray);
+
+    turi::gl_sarray feature_sarray =
+        turi::annotate::featurize_images(image_gl_sarray);
+
+    TS_ASSERT(image_gl_sarray->dtype() == turi::flex_type_enum::IMAGE);
+    TS_ASSERT(feature_sarray.dtype() == turi::flex_type_enum::VECTOR);
+
+    std::vector<turi::flexible_type> feature_vector =
```

Comment:
Use the gl_sarray range iterator here to loop through it.  Tests are a little looser, but this should still be good practice.

Reply:
Done!